### PR TITLE
Version 3

### PR DESCRIPTION
**Pull request by AykutSarac**
## What has changed since version 2?

* New UI
* Edit nodes on graph
* Display path to nodes at Node Modal
* JWT Decoder
* JSON Schema
* Live/Manual Transform toggle
* Performance improvements
* New Landing Page
* Account, Cloud functions, save/share documents
* Fetch JSON through URL
* Image Previews at graph
* Collapse/Expand toggle all nodes
* Display children count on parent nodes

Kudos to @AykutSarac @hanbin9775 @nnamua @liqun98 @roland-becsi-veratrak @cihat @mbrukman @yuvalg99 @Hexrt @dogukanuhn @VladCuciureanu  @shivam1646

image

### CODE REVIEW

File: src/constants/globalStyle.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ const GlobalStyle = createGlobalStyle`
     font-weight: 400;
     font-size: 16px;
     font-family: ${monaSans.style.fontFamily};
-    background: #141517;
+    background: #F1F1F1 !important;
```
Comment (cihat) [inline src/constants/globalStyle.ts:17]: it might be better to remove this `!important` :)

File: src/utils/json/jsonParser.ts [hunk 2]
```diff
@@ -74,6 +74,7 @@ export function parser(jsonStr: string): Graph {
       path: getNodePath(states.graph.nodes, states.graph.edges, node.id),
     }));
 
+    console.log(states.graph);
```
Comment (cihat) [inline src/utils/json/jsonParser.ts:77]: It is good for best practice in removing logs

File: src/utils/json/jsonAdapter.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    return Promise.resolve(json);
+  } catch (error: any) {
+    throw error;
+  }
+};
+
+const jsonToContent = async (json: string, format: FileFormat): Promise<string> => {
+  try {
+    let contents = json;
+    if (!json) return json;
```
Comment (cihat) [inline src/utils/json/jsonAdapter.ts:50]: Would it be better to use switch-case?

**Review by cihat — COMMENTED** [review]
I expressed the small details that I noticed.

File: src/constants/globalStyle.ts (hunk 1 above)
Comment (AykutSarac) [reply src/constants/globalStyle.ts:17]: This overrides mantine's background color, so we can't remove it.

**Comment by cihat** [comment]
I also noticed that there is a decrease in performance compared to the previous version.

image

File: src/utils/json/jsonAdapter.ts (hunk 3 above)
Comment (AykutSarac) [reply src/utils/json/jsonAdapter.ts:50]: I doubt it will make the code more readable.

File: src/utils/json/jsonParser.ts (hunk 2 above)
Comment (AykutSarac) [reply src/utils/json/jsonParser.ts:77]: Nice catch, it shouldn't be there!

**Pull request merged**